crates/ott-balancer/src/monolith.rs

```diff
@@ -149,7 +149,7 @@ pub struct Room {
     metadata: Option<RoomMetadata>,
 
     /// The Sender used to broadcast to all clients in this room.
-    broadcast_tx: tokio::sync::broadcast::Sender<SocketMessage>,
+    broadcast_tx: Arc<tokio::sync::broadcast::Sender<SocketMessage>>,
```

Comment:
This is not what I'm looking for. The objective is to reduce the size of memory allocations for clones on messages that gets passed through the channel. For every receiver on this broadcast channel, the message _must_ be cloned for memory safety, as indicated [in the broadcast channel docs here](https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html).

What you've done here is effectively a no-op. It should be `Sender<Arc<SocketMessage>>`. `Arc` is short for an atomic reference counter. This seems like a good article if you want to learn more: https://reintech.io/blog/understanding-implementing-rust-arc-mutex

Reply:
I have one more type mismatch I can't figure out how to fix. It's in `client.rs` in the `client_entry()` fn. 

```
loop {
        tokio::select! {
            Ok(msg) = client_link.outbound_recv() => {
                if let SocketMessage::Message(msg) = msg {
                    if let Err(err) = stream.send(msg).await {
                        error!("Error sending ws message to client: {:?}", err);
                        break;
                    }
                } else {
                    info!("Client outbound stream ended");
                    break;
                }
            }

            Some(msg) = stream.next() => {
                if let Ok(msg) = msg {
                    if let Err(err) = client_link.inbound_send(msg).await {
                        error!("Error sending client message to balancer: {:?}", err);
                        break;
                    }
                } else {
                    debug!("Client inbound websocket stream ended");
                    break;
                }
            }

            else => {
                debug!("Client websocket stream ended");
                break;
            }
        }
    }

    info!("ending client connection");

    Ok(())
}
```

In the `if let SocketMessage::Message(msg) = msg {` line how can I make `SocketMessage::Message(msg)` a `Arc<SocketMessage>`?

Here's the full error:
```
188 |                 if let SocketMessage::Message(msg) = msg {
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^   --- this expression has type `Arc<messages::SocketMessage>`
    |                        |
    |                        expected `Arc<SocketMessage>`, found `SocketMessage`
    |
    = note: expected struct `Arc<messages::SocketMessage>`
                 found enum `messages::SocketMessage`
```

Reply:
Make this a draft and push what you have so I can play with it